README.md

```diff
@@ -27,34 +27,25 @@ The Python wrapper automatically compiles the C++ library (cirronlib.cpp) on fir
 ```
 from cirron import Collector
 
-collector = Collector()
-
 # Start collecting performance metrics
-collector.start()
-
-# Your code here
-# ...
+with Collector() as collector:
+    # Your code here
+    # ...
 
-# Stop collecting and retrieve the metrics
-metrics = collector.end()
-print(metrics)
+# Retrieve the metrics
+print(collector.counters)
 ```
 
 ### Syscalls
 ```
 from cirron import Tracer, to_tef
 
-tracer = Tracer()
-
-# Start syscall trace
-tracer.start()
-
-# Your code here
-# ...
+with Tracer() as tracer:
+    # Your code here
+    # ...
 
 # Stop collecting and retrieve the trace
-trace = tracer.end()
-print(trace)
+print(tracer.trace)
 
 # Save the trace for ingesting to Perfetto
 open("/tmp/trace", "w").write(to_tef(trace))
```

Comment:
Doesn't matter match as it's an example, but maybe it could be updated too to `with open('/tmp/trace', 'w') as f: f.write(to_tef(trace))`

Reply:
Oh, actually the `trace` here needs to be updated.

Reply:
There could even have an example with two context managers too, e.g.

```py
with Tracer() as tracer, open('/tmp/trace', 'w') as f:
  # your code here
  trace = tracer.trace

  # Save the trace for ingesting to Perfetto
  f.write(to_tef(trace))
```

Or keep both examples separated, either way is fine I think :+1: